employee-rostering-docs/src/main/asciidoc/ProjectStructure/ProjectStructure-section.adoc

```diff
@@ -0,0 +1,236 @@
+[[projectstructure]]
+= Project Structure
+:doctype: book
+:imagesdir: ..
+:sectnums:
+:toc: left
+:icons: font
+:experimental:
+
+The project is structured in the following folders:
+
+* *employee-rostering-backend* core planning domain model and backend REST api implemented with Spring Boot
+* *employee-rostering-benchmark* solution benchmark
+* *employee-rostering-distribution* assembly logic for the deployment assets
+* *employee-rostering-docs* this documentation
+* *employee-rostering-frontend* user interface implemented with ReactJS
+
+== Domain Model
+
+The domain model is the most important piece of a project based on OptaPlanner. A careful design simplifies the
+constraint definition. The classes of the domain model are placed in the *employee-rostering-backend* module.
+
+The most important classes to understand the domain model are:
+
+* *Shift* is the _planning entity_, where is the defined the relationship with the _planning variable_ employee. Other important fields: `spot`, `rotationEmployee`, `startDateTime`, `endDateTime`.
+* *Employee* is the _planning variable_, it's identified by the `name` and has a set of skills (`skillProficiencySet`).
+* *Roster* is the _planning solution_, `employeeList` is the list of employees (the range of values that can be assigned to the _planning variable_), the field `score` holds the score (3 levels: hard, medium, soft), the other problem facts are: `skillList`, `spotList`, `employeeAvailabilityList`, `rosterParametrization`, `rosterState`, `shiftList`.
+
+== Constraints
+
+The constraints are defined in the *employee-rostering-backend* module, with the implementation of the backend REST
+service.
+
+*Worth noting:*
+
+* The solver configuration file:
+`employee-rostering-backend/src/main/resources/org/optaweb/employeerostering/service/solver/employeeRosteringSolverConfig.xml`
+* The constraints definition file: `employee-rostering-backend/src/main/resources/org/optaweb/employeerostering/service/solver/employeeRosteringScoreRules.drl`
+
+=== Constraint definition
+
+The constraints are defined using the DRL language. See: https://docs.optaplanner.org/latestFinal/optaplanner-docs/html_single/index.html#implementingAScoreRule[Implementing a score rule].
+
+==== Hard Constraints
+
+*Required skill for a shift*
+
+[source,ruby]
+----
+rule "Required skill for a shift"
+    when
+        Shift(
+                employee != null,
+                !getEmployee().hasSkills(getSpot().getRequiredSkillSet()))
+    then
+        scoreHolder.addHardConstraintMatch(kcontext, -100);
+end
```

Comment:
Do we want to include all DRL implementations in the documentation?
This increases the maintenance burden: every constraint change must now be duplicated into the docs.

I don't think the docs reader needs to see the DRL. An explanation of the constraint should be enough. Wdyt?

Reply:
True, but docs on certain DRL rules might make the reader more comfortable in learning how to modify the example to include their rule set.